src/integrator/cuda/cuda_eval_denvars.cu

```diff
@@ -77,44 +82,62 @@ __global__ void eval_uvars_gga_kernel( size_t           ntasks,
 
   const auto* den_basis_prod_device = task.zmat;
 
-  const int tid_x = blockIdx.x * blockDim.x + threadIdx.x;
-  const int tid_y = blockIdx.y * blockDim.y + threadIdx.y;
-
-  register double den_reg = 0.;
-  register double dx_reg  = 0.;
-  register double dy_reg  = 0.;
-  register double dz_reg  = 0.;
-
-  if( tid_x < nbf and tid_y < npts ) {
-
-    const double* bf_col   = basis_eval_device     + tid_y*nbf;
-    const double* bf_x_col = dbasis_x_eval_device  + tid_y*nbf;
-    const double* bf_y_col = dbasis_y_eval_device  + tid_y*nbf;
-    const double* bf_z_col = dbasis_z_eval_device  + tid_y*nbf;
-    const double* db_col   = den_basis_prod_device + tid_y*nbf;
-
-    den_reg = bf_col[ tid_x ]   * db_col[ tid_x ];
-    dx_reg  = bf_x_col[ tid_x ] * db_col[ tid_x ];
-    dy_reg  = bf_y_col[ tid_x ] * db_col[ tid_x ];
-    dz_reg  = bf_z_col[ tid_x ] * db_col[ tid_x ];
-
+  __shared__ double den_shared[4][GGA_KERNEL_SM_BLOCK_X][GGA_KERNEL_SM_BLOCK_Y+1];
```

Comment:
I think we can switch these over to `warp_size` and `max_warps_per_thread_block` from 
https://github.com/wavefunction91/GauXC/blob/master/src/integrator/cuda/cuda_device_properties.hpp#L7